### PR TITLE
Feature/GAL-1193

### PR DESCRIPTION
**Pull request by MichaelW-NHS**
## Description

https://nhsd-jira.digital.nhs.uk/browse/GAL-1193
- Added additional validation on the validateClinicCapacity lambda, checking there are exactly six weeks of schedules and that the week commencing date always starts on a Monday.
- Additional unit tests and test data amendments for the above.
 
## Type of changes

- [ ] Refactoring (non-breaking change)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [X] I have followed the code style of the project
- [X] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [X] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/validateClinicCapacityLambda/validateClinicCapacityLambda.js [hunk 1]
```diff
@@ -102,6 +104,22 @@ export async function validateRecord(record, client) {
         validationResults.message = `Invalid JSON Schema`;
         console.error("errors : ", validation.errors);
         return validationResults;
+      } else {
+        record.ClinicScheduleSummary.ClinicScheduleSummary.forEach((clinic) => {
+          if(clinic.Schedule.length !== SIX_WEEK_HORIZON) {
```
Comment (chenglawson1) [inline terraform/src/validateClinicCapacityLambda/validateClinicCapacityLambda.js:109]: I think the AC just says it cannot exceed 6 weeks, not it has to be exactly 6 weeks. The example given by Grail only has 3 weeks for each clinic.

Comment (chenglawson1) [reply terraform/src/validateClinicCapacityLambda/validateClinicCapacityLambda.js:109]: Please clarify with Oyin.

Comment (chenglawson1) [reply terraform/src/validateClinicCapacityLambda/validateClinicCapacityLambda.js:109]: I've just checked with Tony, his understanding is Grail will always send 6 weeks' schedule, so if any week has no capacity they will send 0. We will confirm this with them again tomorrow.

Comment (MichaelW-NHS) [reply terraform/src/validateClinicCapacityLambda/validateClinicCapacityLambda.js:109]: Thanks. I'll for tomorrow then. If it's confirmed then is the added validation fine?

Comment (chenglawson1) [reply terraform/src/validateClinicCapacityLambda/validateClinicCapacityLambda.js:109]: Grail has confirmed that each clinic will have 6 weeks's schedule in the message, so no code change required.

**Review by chenglawson1 — APPROVED** [review]

**Review by NeerjaGoyalNHS — APPROVED** [review]

**Pull request merged**